app/controllers/Search.scala

```diff
@@ -48,4 +56,92 @@ object Search extends Controller {
       }
     }
   }
+
+  def openSearchDescription(dtype: String) = Action { implicit request =>
+    def templateUrl(ftype: String) = {
+      routes.Search.index.absoluteURL + "/results?q={searchTerms}&page={startPage?}&perpage={count?}" +
+      "&target=" + dtype + "&format=" + ftype
+    }
+    Ok(
+      <OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
```

Comment:
In my opinion, ideally yes.

But honestly, the template generation is enough of a pain in the ass in Play that I'll definitely forgive you for choosing to leave this here. But keep in mind I just might use this as an excuse for generating HTML in Controllers and dropping Twirl entirely ;) [Don't worry, that won't happen even if I kinda want to]. I'd suggest we leave this here for now and refactor all the generated XML to templates if it makes sense later.